interface/ceed-vector.c

```diff
@@ -238,6 +242,50 @@ int CeedVectorCopy(CeedVector vec, CeedVector vec_copy) {
   return CEED_ERROR_SUCCESS;
 }
 
+/**
+  @brief Copy a strided portion of `CeedVector` contents into a different `CeedVector`
+
+  @param[in]     vec      `CeedVector` to copy
+  @param[in]     start    First index to copy
+  @param[in]     step     Stride between indices to copy
+  @param[in,out] vec_copy `CeedVector` to copy values to
+
+  @return An error code: 0 - success, otherwise - failure
+
+  @ref User
+**/
+int CeedVectorCopyStrided(CeedVector vec, CeedSize start, CeedInt step, CeedVector vec_copy) {
```

Comment:
Do we want to check that the length of `vec` is evenly divisible by `step`? Might be too constrictive, but seems a useful debugging feature to me at least.

Reply:
Looks like this check would break the test you added (length of 10, but step of 3).

Are there times when we would expect the length to not be evenly divisible by `step` when using these strided routines?

Reply:
I don't think we need to care that the length of the vec is a multiple of the step?